tsdb/engine/tsm1/wal.go

```diff
@@ -250,12 +254,18 @@ func (l *Log) WritePoints(points []models.Point, fields map[string]*tsdb.Measure
 
 // addToCache will add the points, measurements, and fields to the cache and return true if successful. They will be queryable
 // immediately after return and will be flushed at the next flush cycle. Before adding to the cache we check if we're over the
-// max memory threshold. If we are we request a flush in a new goroutine and return false, indicating we didn't add the values
+// max memory threshold. If we are we request a flush in a new goroutine and return an error, indicating we didn't add the values
 // to the cache and that writes should return a failure.
-func (l *Log) addToCache(points []models.Point, fields map[string]*tsdb.MeasurementFields, series []*tsdb.SeriesCreate, checkMemory bool) bool {
+func (l *Log) addToCache(points []models.Point, fields map[string]*tsdb.MeasurementFields, series []*tsdb.SeriesCreate, checkMemory bool) error {
 	l.cacheLock.Lock()
 	defer l.cacheLock.Unlock()
 
+	select {
```

Comment:
I could here, but in `writeToLog`, I would have to take a acquire the `cacheLock` at the beginning of the func to avoid a race.  That func only acquires a `writeLock` and acquires the `cacheLock` later.  I didn't want to move the lock ordering around though and this keeps the closing/closed checks consistent.